### ISSUE TITLE
Revert the PR that killed CustomizeUI extension

### ISSUE THREAD

**Issue by MuhsinFatih** [microsoft/vscode#181298]
The PR #166126 killed the extremely popular extension [CustomizeUI](https://github.com/iocave/customize-ui) (~160k downloads) a while ago ([discussion here](https://github.com/iocave/customize-ui/issues/156)). There was a lot of backlash from the community in the PR comments, but the PR was merged and I can't see any issues mentioning this problem right now.
Customizing the UI was such a fundamental feature of vscode and I believe most people would agree that it's much more important than cutting down 1.8MB of bundle size. For me personally, CustomizeUI saved a lot of real estate by:
- moving the activity bar icons to a horizontal setting
- hiding the top bar
- making the tab bar shorter

The difference in real estate is so much that I've been considering switching entirely to nvim for this reason alone. I'd love to be able to customize my vscode like before.

Are there any plans to allow fully customizing the UI through some other means, or to simply revert the PR changes?

**Comment by JakeSays** [comment]
Yeah 1.8MB is NOTHING compared to UI customization. NOTHING. PLEASE revert the PR.

**Comment by jrieken** [comment]
I have locked this issue as for being too heated. 

We have explained our reasoning and background here https://github.com/microsoft/vscode-discussions/discussions/257#discussioncomment-4788068

To summarise: nothing has been killed, there is no evil intent, what CustomizeUI did is still possible, and us (the people reading these comments) are just people like you. 

Extensions like CustomizeUI extend VS Code not by using the API but by applying reverse-engineered patches. These patches break whenever something "inside" VS Code changes. We tolerate extensions taking such an approach but we can support them (it would me we cannot change VS Code any further). Doing private property mangling was a lot of those changes but fundamentally patches still work, we also publish source maps that include the reverse mapping from mangles to original name. So, anyone with constructive energy can step up and fix CustomizeUI, we are happy to help with source maps but we won't revert mangling.
